diff --git a/src/date/dateparser-inl.h b/src/date/dateparser-inl.h
index c10163b..1296af4 100644
--- a/src/date/dateparser-inl.h
+++ b/src/date/dateparser-inl.h
@@ -340,8 +340,8 @@
   // Successfully parsed ES5 Date Time String.
   // ES#sec-date-time-string-format Date Time String Format
   // "When the time zone offset is absent, date-only forms are interpreted
-  //  as a UTC time and date-time forms are interpreted as a local time."
-  if (tz->IsEmpty() && time->IsEmpty()) {
+  //  as a UTC time and date-time forms are also should be interpreted as a UTC time, just like JSC"
+  if (tz->IsEmpty()) {
     tz->Set(0);
   }
   day->set_iso_date();
